kotlin-result-coroutines/src/commonTest/kotlin/com.github.michaelbull.result.coroutines/binding/SuspendableBindingTest.kt

```diff
@@ -100,6 +101,90 @@ class SuspendableBindingTest {
         }
     }
 
+    @Test
+    fun returnsStateChangedUntilFirstBindFailedWhenBindingSetToNotCancelEagerly() {
+        var xStateChange = false
+        var yStateChange = false
+        var zStateChange = false
+        suspend fun provideX(): Result<Int, BindingError> {
+            delay(1)
+            xStateChange = true
+            return Ok(1)
+        }
+
+        suspend fun provideY(): Result<Int, BindingError> {
+            delay(10)
+            yStateChange = true
+            return Err(BindingError)
+        }
+
+        suspend fun provideZ(): Result<Int, BindingError> {
+            delay(1)
+            zStateChange = true
+            return Err(BindingError)
+        }
+
+        runBlockingTest {
+            val result = binding<Int, BindingError>(eagerlyCancel = false) {
+                val x = provideX().bind()
+                val y = provideY().bind()
+                val z = provideZ().bind()
+                x + y + z
+            }
+
+            assertTrue(result is Err)
+            assertEquals(
+                expected = BindingError,
+                actual = result.error
+            )
+            assertTrue(xStateChange)
+            assertTrue(yStateChange)
+            assertFalse(zStateChange)
+        }
+    }
+
+    @Test
+    fun returnsStateChangedUntilFirstBindFailedWhenBindingSetToCancelEagerly() {
+        var xStateChange = false
+        var yStateChange = false
+        var zStateChange = false
+        suspend fun provideX(): Result<Int, BindingError> {
+            delay(1)
+            xStateChange = true
+            return Ok(1)
+        }
+
+        suspend fun provideY(): Result<Int, BindingError> {
+            delay(10)
+            yStateChange = true
+            return Err(BindingError)
+        }
+
+        suspend fun provideZ(): Result<Int, BindingError> {
+            delay(1)
+            zStateChange = true
+            return Err(BindingError)
+        }
+
+        runBlockingTest {
```

Comment:
I guess we need a `return` statement on this line given our discussion in the most recent PR.

Reply:
sure, If the other pr is merged first I will pull and fix these, otherwise maybe better to do in that pr.